mod/badges/set/abstract/award_badges.rb

```diff
@@ -20,10 +20,14 @@ def earns_badge action
 
 def award_badge badge_card
   name_parts = [Auth.current, badge_card.badge_type, :badges_earned]
-  badge_pointer =
-    subcard(name_parts) ||
-      attach_subcard(Card.fetch(name_parts, new: { type_id: PointerID }))
+  badge_pointer = Card.fetch(name_parts, new: { type_id: PointerID })
+  if ActManager.include? badge_pointer.name
+    director = ActManager.fetch(badge_pointer)
+    director.reset_stage
+    badge_pointer = director.card
+  end
```

Comment:
I'm not sure if `badge_pointer = director.card` is really necessary. I think I added that before I found the real issue which is that on metric value imports we add all metric values in the integrate stage of the import card. When the first metric value adds a badge to the badge_pointer the badge_pointer catches up to the integrate stage and all badges added later to the badge_pointer by other metric values of that import are not saved (because the storage phase for the badge_pointer is already completed)